### PR TITLE
Add `leo report` hooks on the user script

### PR DESCRIPTION
**Pull request by 0x2b3bfa0**
Not tested, may need heavy reformatting. To be cascaded into #733.

#### Statuses

* `queued` — waiting for a new spot instance
* `running` — the user-specified script started to run
* `timeout` — terminated after exceeding the execution time limit
* `succeeded` or `failed` — finished voluntarily with [non-]zero exit status

> **Note** when a new task is created, its status should be hardcoded to `queued` until machines submit their first status report.

### CODE REVIEW

File: task/common/machine/testdata/machine_script_minimal.golden [hunk 1]
```diff
@@ -48,7 +48,7 @@ sudo tee /etc/systemd/system/tpi-task.service > /dev/null <<END
 [Service]
   Type=simple
   ExecStart=-$TPI_START_COMMAND
-  ExecStop=/bin/bash -c 'source /opt/task/credentials; systemctl is-system-running | grep stopping || echo "{\\\\"result\\\\": \\\\"\$SERVICE_RESULT\\\\", \\\\"code\\\\": \\\\"\$EXIT_STATUS\\\\", \\\\"status\\\\": \\\\"\$EXIT_CODE\\\\"}" > "$TPI_LOG_DIRECTORY/status-$TPI_MACHINE_IDENTITY" && RCLONE_CONFIG= rclone copy "$TPI_LOG_DIRECTORY" "\$RCLONE_REMOTE/reports"'
+  ExecStop=/bin/bash -c 'source /opt/task/variables; source /opt/task/credentials; if systemctl is-system-running | grep stopping; then status=queued; else echo "{\\\\"result\\\\": \\\\"\$SERVICE_RESULT\\\\", \\\\"code\\\\": \\\\"\$EXIT_STATUS\\\\", \\\\"status\\\\": \\\\"\$EXIT_CODE\\\\"}" > "$TPI_LOG_DIRECTORY/status-$TPI_MACHINE_IDENTITY" && RCLONE_CONFIG= rclone copy "$TPI_LOG_DIRECTORY" "\$RCLONE_REMOTE/reports"; if test $SERVICE_RESULT == timeout; then status=timeout; else; test $EXIT_STATUS == 0 && status=succeeded || status=failed; fi; fi && leo report --type=data --url $STUDIO_URL --token $STUDIO_TOKEN --repo $STUDIO_REPO_URL --param status=$status'
```
Comment (tasdomas) [inline task/common/machine/testdata/machine_script_minimal.golden:51]: Shouldn't `\\\\"` be `\\\"`?

Comment (0x2b3bfa0) [reply task/common/machine/testdata/machine_script_minimal.golden:51]: I don't recall exactly why, but this is the right amount of backslashes and it works. 😅

Comment (0x2b3bfa0) [reply task/common/machine/testdata/machine_script_minimal.golden:51]: They should never be an odd number, though.

Comment (tasdomas) [reply task/common/machine/testdata/machine_script_minimal.golden:51]: Going through levels of escaping:
`"` -> `\`+`"` -> `\\` + `\"`

Comment (tasdomas) [reply task/common/machine/testdata/machine_script_minimal.golden:51]: Either way, if we're having discussions about how many backslashes we need to escape this, they are beside the point.

**Comment by 0x2b3bfa0** [comment]
Closing along with #733, superseded by #735 🙃

**Pull request closed without merging**